src/librustdoc/html/static/main.js

```diff
@@ -282,6 +284,14 @@ function getSearchElement() {
     function expandSection(id) {
         var elem = document.getElementById(id);
         if (elem && isHidden(elem)) {
+            if (elem.tagName === "CODE" && elem.parentNode.tagName === "H4") {
+                // We are in a trait implementation, what we want is the parent.
```

Comment:
Would it not be better to add a class or somesuch in the generated code so that we're not relying so hard on the exact element structure for this?

Reply:
The problem is that it would require a way bigger change to simplify the DOM in order to do so. At first I thought about adding an id to the collapse button but in this case, we have both the `<code>` and the `<h2>` with ids, making it impossible. This is why I change the page's hash when it's the `<code>`.

Reply:
How does the id get into the code/h2 ?  if it's purely via the markdown parser then I suppose I can understand the difficulty.  If it's in rustdoc though then we ought to be able to do something.  Failing that, the rest looks okay and if you're sure it's reliable then you can r=me.

Reply:
It's worth checking. I'll take another look around to be sure about it.